activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -140,6 +140,14 @@ def full_version
         def get_full_version
           @connection.server_info[:version]
         end
+
+        def translate_exception(exception, message:, sql:, binds:)
+          if !exception.error_number && exception.is_a?(Mysql2::Error::TimeoutError)
```

Comment:
AFAICS this could be any StandardError, so I don't think it's safe to assume it'll respond to `error_number` -- maybe just flip these two conditions around?